beacon-chain/core/electra/validator.go

```diff
@@ -0,0 +1,103 @@
+package electra
+
+import (
+	"context"
+	"errors"
+
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+)
+
+// SwitchToCompoundingValidator
+//
+// Spec definition:
+//
+//	 def switch_to_compounding_validator(state: BeaconState, index: ValidatorIndex) -> None:
+//		validator = state.validators[index]
+//		if has_eth1_withdrawal_credential(validator):
+//		    validator.withdrawal_credentials = COMPOUNDING_WITHDRAWAL_PREFIX + validator.withdrawal_credentials[1:]
+//		    queue_excess_active_balance(state, index)
+func SwitchToCompoundingValidator(ctx context.Context, s state.BeaconState, idx primitives.ValidatorIndex) error {
+	v, err := s.ValidatorAtIndex(idx)
+	if err != nil {
+		return err
+	}
+	if len(v.WithdrawalCredentials) == 0 {
+		return errors.New("validator has no withdrawal credentials")
```

Comment:
Are you sure this should be an error and not just `return nil`? I think it is valid for a consolidation to come from a validator where `has_compounding_withdrawal_credential` is true and `has_eth1_withdrawal_credential` is false. I think this would be the case where you decide to consolidate 2 validators that were created with compounding withdrawal credentials? But this function would treat the presence of such a pending consolidation as an error.

Reply:
Strike that, I see this is protecting against an out of bound index panic. It should be impossible for a validator without a credential to even get in the consolidations queue. Makes sense to be defensive here even though something is deeply messed up if we hit this case.